msal/src/Microsoft.Identity.Client/Features/DeviceCode/DeviceCodeResult.cs

```diff
@@ -0,0 +1,101 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// This object is returned as part of the device code flow
+    /// and has information intended to be shown to the user about
+    /// where to navigate to login and what the device code is that needs
+    /// to be entered.
+    /// </summary>
+    public class DeviceCodeResult
+    {
+        internal DeviceCodeResult(
+            string userCode,
+            string deviceCode,
+            string verificationUrl,
+            DateTimeOffset expiresOn,
+            long interval,
+            string message,
+            string clientId,
+            string resource)
+        {
+            UserCode = userCode;
+            DeviceCode = deviceCode;
+            VerificationUrl = verificationUrl;
+            ExpiresOn = expiresOn;
+            Interval = interval;
+            Message = message;
+            ClientId = clientId;
+            Resource = resource;
+        }
+
+        /// <summary>
+        /// User code returned by the service
+        /// </summary>
+        public string UserCode { get; }
+
+        /// <summary>
+        /// Device code returned by the service
+        /// </summary>
+        public string DeviceCode { get; }
+
+        /// <summary>
+        /// Verification URL where the user must navigate to authenticate using the device code and credentials.
+        /// </summary>
+        public string VerificationUrl { get; }
+
+        /// <summary>
+        /// Time when the device code will expire.
+        /// </summary>
+        public DateTimeOffset ExpiresOn { get; }
+
+        /// <summary>
+        /// Polling interval time to check for completion of authentication flow.
+        /// </summary>
+        public long Interval { get; }
+
+        /// <summary>
+        /// User friendly text response that can be used for display purpose.
+        /// </summary>
+        public string Message { get; }
+
+        /// <summary>
+        /// Identifier of the client requesting device code.
+        /// </summary>
+        public string ClientId { get; }
+
+        /// <summary>
+        /// Identifier of the target resource that would be the recipient of the token.
+        /// </summary>
+        public string Resource { get; }
+        // todo: this should be SCOPE(S) and not resource for v2 endpoint...

```

Comment:
want to just rename it already now?

Reply:
I'll fix this.  I need to debug this case to look at what comes back for it in the /organizations flow or if this is just the /scopes that the user submitted in the first place.